pkg/controller/deployment/deployment_controller.go

```diff
@@ -84,8 +91,63 @@ func (d *DeploymentController) reconcileDeployment(deployment *extensions.Deploy
 }
 
 func (d *DeploymentController) reconcileRecreateDeployment(deployment extensions.Deployment) error {
-	// TODO: implement me.
-	return nil
+	oldRCs, err := d.getOldRCs(deployment)
+	if err != nil {
+		return err
+	}
+
+	// kill all existing pods for old rcs
+	waitScale := sync.WaitGroup{}
+	waitScale.Add(len(oldRCs))
+	for _, oldRC := range oldRCs {
+		go func(rc *api.ReplicationController) {
+			defer waitScale.Done()
+			if _, err := d.scaleRCAndRecordEvent(rc, 0, deployment); err != nil {
+				glog.Errorf("Fail to scale replication controller: %q/%q", rc.Namespace, rc.Name)
+				util.HandleError(err)
+			}
+		}(oldRC)
+	}
+	waitScale.Wait()
+
+	// waite for all pods killed, then delete all old rcs
+	if err := wait.Poll(Interval, Timeout, client.ControllerHasDesiredReplicas(d.client, oldRCs[0])); err != nil {
```

Comment:
Shouldn't you do this for all oldRCs? (you could use a wait group to do it in parallel)